lib/masamune/schema/map.rb

```diff
@@ -24,6 +24,34 @@
 
 module Masamune::Schema
   class Map
+    class JSONEncoder < SimpleDelegator
+      def initialize(io, store)
+        super io
+        @store = store
+      end
+
+      def gets(*a)
+        line = __getobj__.gets(*a)
+        return unless line
+        return line if skip?
+        encode(line.split(separator)).join(separator)
+      end
+
+      private
+
+      def skip?
+        @store.json_encoding == :quoted
+      end
+
+      def encode(fields = [])
+        fields.map { |field| field =~ /^{|}$/ ? %Q{"#{field.gsub('"', '""')}"} : field }
```

Comment:
double quotes in CSV are [escaped using two double quote characters](http://en.wikipedia.org/wiki/Comma-separated_values#Toward_standardization):
> A (double) quote character in a field must be represented by two (double) quote characters.

there is test coverage demonstrating that this is valid CSV encoding